app/models/user.rb

```diff
@@ -16,9 +18,24 @@ class User < ApplicationRecord
   }
 
   has_one :member
+  has_many :organization_members, dependent: :destroy
+  has_many :organizations, through: :organization_members
+
+  # Adapted from https://github.com/heartcombo/devise/blob/main/lib/devise/models/validatable.rb
+  # so we can scope email uniqueness to library_id
+  validates :email, presence: {if: :email_required?}
+  validates :email, uniqueness: {allow_blank: true, case_sensitive: false, if: :devise_will_save_change_to_email?, scope: :library_id}
```

Comment:
This validation is why we needed to drop `:validatable` above. `:validatable` adds this validation without the `scope: :library_id`